dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/BaseDimensionalObject.java

```diff
@@ -256,14 +256,14 @@ public String getKey()
     {
         QueryKey key = new QueryKey();
 
-        key.add( getDimension() );
-        getItems().forEach( e -> key.add( e.getDimensionItem() ) );
+        key.add( "dimension", getDimension() );
```

Comment:
My only comment is regarding all the strings used through these changes.
Maybe we should use enums or constants to make them more consistent and easier to maintain.

Reply:
Right. The only issue is that we would need one for each property of data query and event query objects. It would tightly couple the QueryKey directly to the analytics objects. I think it is more viable to decouple it and accept strings.

Reply:
Revolving the conversation. No changes required.